test/pkg_finish_test.rb

```diff
@@ -0,0 +1,182 @@
+    end
+
+    it "saves repository information" do
+      expect(Yast::Pkg).to receive(:SourceSaveAll)
+      expect(Yast::Pkg).to receive(:TargetFinish)
+      expect(Yast::Pkg).to receive(:SourceCacheCopyTo).with(destdir)
+      allow(Yast::WFM).to receive(:Execute)
+      expect(client.run).to be_nil
+    end
+
+    it "copies failed_packages list under destination dir" do
+      stub_const("Yast::Pkg", double("pkg").as_null_object)
+      expect(Yast::WFM).to receive(:Execute).
+        with(Yast::Path.new(".local.bash"),
+        "test -f /var/lib/YaST2/failed_packages && "\
+        "/bin/cp -a /var/lib/YaST2/failed_packages '#{destdir}/var/lib/YaST2/failed_packages'")
+      client.run
+    end
+
+    context "given CD/DVD repositories" do
+      let(:repositories) { [cd_repo, http_repo] }
+
+      let(:cd_repo) do
+        Packages::Repository.new(repo_id: 1, name: "SLE-12-SP2-0", enabled: true,
+          url: URI("cd://dev/sr0"), autorefresh: false)
+      end
+
+      let(:http_repo) do
+        Packages::Repository.new(repo_id: 2, name: "SLE-12-SP2-Pool", enabled: true,
+          url: URI("http://download.suse.com/sle-12-sp2"), autorefresh: true)
+      end
+
+      let(:sles_product) do
+        Packages::RepositoryProduct.new(name: "SLES", version: "12.2",
+          arch: "x86_64", category: "base", status: :available, vendor: "SUSE")
+      end
+
+      let(:sles_ha_product) do
+        Packages::RepositoryProduct.new(name: "SLESHA", version: "12.2",
+          arch: "x86_64", category: "base", status: :available, vendor: "SUSE")
+      end
+
+      before do
+        allow(cd_repo).to receive(:products).and_return([sles_product.clone])
+      end
+
+      context "if their products are available through other repos" do
+        before do
+          allow(http_repo).to receive(:products).and_return([sles_product.clone])
+        end
+
+        it "disables those repositories" do
+          expect(cd_repo).to receive(:disable!)
+          client.run
+        end
+      end
+
+      context "if their products are not available through other repos" do
+        before do
+          allow(http_repo).to receive(:products).and_return([sles_ha_product])
+        end
+
+        it "does not disabled those repositories" do
+          expect(cd_repo).to_not receive(:disable!)
+          client.run
+        end
+      end
+    end
+
+    context "during update" do
+      let(:update) { true }
+      let(:tmpdir) { Pathname.new(__FILE__).dirname.join("tmp") }
+      let(:repos_dir) { tmpdir.join("repos.d") }
+      let(:vardir) { tmpdir.join("var") }
+
+      before do
+        allow(Yast::Directory).to receive(:vardir).and_return(vardir.to_s)
+        allow(Yast::WFM).to receive(:call)
+        stub_const("Yast::PkgFinishClient::REPOS_DIR", repos_dir.to_s)
+        stub_const("Yast::Pkg", double("pkg").as_null_object)
+      end
+
+      around do |example|
+        FileUtils.rm_rf(tmpdir)
```

Comment:
NP: ... then you could remove this cleanup at the beginning, the `mktmpdir` ensures it's a new dir (without any possible leftovers from previous tests or runs).